test/trace_api_test.c

```diff
@@ -67,15 +67,15 @@ static int test_trace_categories(void)
 }
 
 #ifndef OPENSSL_NO_TRACE
-static void put_trace_output()
+static void put_trace_output(void)
```

Comment:
Please do not put unrelated changes into a PR.

Reply:
I was compelled to include this because otherwise my local builds fail with:
```
test/trace_api_test.c:70:13: error: function declaration isn't a prototype [-Werror=strict-prototypes]
   70 | static void put_trace_output()
      |             ^~~~~~~~~~~~~~~~
test/trace_api_test.c:78:12: error: function declaration isn't a prototype [-Werror=strict-prototypes]
   78 | static int test_trace_channel()
      |            ^~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
```
since I generally configure with `--strict-warnings -std=gnu90`.
I fear that some CI part(s) will also fail without those two fixes.

Reply:
Rebased on the meanwhile merged PR #19277, so that commit is no more part of this PR.